src/services/OpenAPIParser.ts

```diff
@@ -335,6 +336,98 @@ export class OpenAPIParser {
     return receiver;
   }
 
+  derefSchemaWithExample(
```

Comment:
I have reused the existing functions in their parser, such as `deref` and `mergeAllOf` which have their own recursion limits set in place by the constant `MAX_DEREF_DEPTH`, and it is currently set to 999. I have also ensured not to recurse a property if it doesn't have a corresponding example, and especially if it references a parent ref to avoid an infinite loop.